scene/resources/portable_compressed_texture.cpp

```diff
@@ -180,22 +181,45 @@ void PortableCompressedTexture2D::create_from_image(const Ref<Image> &p_image, C
 		} break;
 		case COMPRESSION_MODE_S3TC:
 		case COMPRESSION_MODE_ETC2:
-		case COMPRESSION_MODE_BPTC: {
-			ERR_FAIL_COND(p_image->is_compressed());
+		case COMPRESSION_MODE_BPTC:
+		case COMPRESSION_MODE_ASTC: {
 			encode_uint16(DATA_FORMAT_IMAGE, buffer.ptrw() + 2);
-			Ref<Image> copy = p_image->duplicate();
-			switch (p_compression_mode) {
-				case COMPRESSION_MODE_S3TC:
-					copy->compress(Image::COMPRESS_S3TC);
-					break;
-				case COMPRESSION_MODE_ETC2:
-					copy->compress(Image::COMPRESS_ETC2);
-					break;
-				case COMPRESSION_MODE_BPTC:
-					copy->compress(Image::COMPRESS_BPTC);
-					break;
-				default: {
-				};
+			Ref<Image> copy = p_image;
+			if (p_image->is_compressed()) {
+				switch (p_compression_mode) {
+					case COMPRESSION_MODE_S3TC:
+						ERR_FAIL_COND(!(p_image->get_format() >= Image::FORMAT_DXT1 && p_image->get_format() <= Image::FORMAT_RGTC_RG) && p_image->get_format() != Image::FORMAT_DXT5_RA_AS_RG);
```

Comment:
These should probably have user friendly messages as the condition itself isn't easy to parse for end users. Since it's something they can trigger by calling the API manually, it should have a human readable message (with `ERR_FAIL_COND_MSG`).